test/v5/createAmendForeignPropertyAnnualSubmission/CreateAmendForeignPropertyAnnualSubmissionValidatorFactorySpec.scala

```diff
@@ -0,0 +1,105 @@
+import mocks.MockAppConfig
+import play.api.libs.json._
+import support.UnitSpec
+import v5.createAmendForeignPropertyAnnualSubmission.def1.Def1_CreateAmendForeignPropertyAnnualSubmissionValidator
+import v5.createAmendForeignPropertyAnnualSubmission.model.request.CreateAmendForeignPropertyAnnualSubmissionRequestData
+
+class CreateAmendForeignPropertyAnnualSubmissionValidatorFactorySpec extends UnitSpec with MockAppConfig {
+
+  private val validNino       = "AA123456A"
+  private val validBusinessId = "XAIS12345678901"
+  private val validTaxYear    = "2023-24"
+
+  private val validStructuredBuildingAllowance =
+    Json.parse("""{
+        |  "amount": 3545.12,
+        |  "firstYear": {
+        |    "qualifyingDate": "2020-03-29",
+        |    "qualifyingAmountExpenditure": 3453.34
+        |  },
+        |  "building": {
+        |    "name": "Building name",
+        |    "number": "12",
+        |    "postcode": "TF3 4GH"
+        |  }
+        |}""".stripMargin)
+
+  private def entryWith(countryCode: String, structuredBuildingAllowance: JsValue*) =
+    Json.parse(s"""
+                  |    {
+                  |      "countryCode": "$countryCode",
+                  |      "adjustments": {
+                  |        "privateUseAdjustment": 4553.34,
+                  |        "balancingCharge": 3453.34
+                  |      },
+                  |      "allowances": {
+                  |        "annualInvestmentAllowance": 38330.95,
+                  |        "costOfReplacingDomesticItems": 41985.17,
+                  |        "zeroEmissionsGoodsVehicleAllowance": 9769.19,
+                  |        "otherCapitalAllowance": 1049.21,
+                  |        "electricChargePointAllowance": 3565.45,
+                  |        "structuredBuildingAllowance": ${JsArray(structuredBuildingAllowance)},
+                  |        "zeroEmissionsCarAllowance": 3456.34
+                  |      }
+                  |    }""".stripMargin)
+
+  private def bodyWith(nonFhlEntries: JsValue*) = Json.parse(
+    s"""{
+       |  "foreignFhlEea": {
+       |    "adjustments": {
+       |      "privateUseAdjustment": 34343.45,
+       |      "balancingCharge": 53543.23,
+       |      "periodOfGraceAdjustment": true
+       |    },
+       |    "allowances": {
+       |      "annualInvestmentAllowance": 3434.23,
+       |      "otherCapitalAllowance": 1343.34,
+       |      "electricChargePointAllowance": 6565.45,
+       |      "zeroEmissionsCarAllowance": 3456.34
+       |    }
+       |  },
+       |  "foreignNonFhlProperty": ${JsArray(nonFhlEntries)}
+       |}
+       |""".stripMargin
+  )
+
+  private val validBody = bodyWith(entryWith("AFG", validStructuredBuildingAllowance))
+
+  private val validatorFactory = new CreateAmendForeignPropertyAnnualSubmissionValidatorFactory(mockAppConfig)
+
+  private def setUpMocks(): Unit = MockedAppConfig.minimumTaxV2Foreign.returns(TaxYear.starting(2021)).anyNumberOfTimes()
+
```

Comment:
Most of the above set-up isn't really relevant to what is being tested. See the latest multi-schema how-to guide which has an example of testing validator factories (or see https://github.com/hmrc/self-assessment-bsas-api/pull/519)